src/python_testing/TC_DGSW_2_2.py

```diff
@@ -0,0 +1,156 @@
+#
+#    Copyright (c) 2024 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+# See https://github.com/project-chip/connectedhomeip/blob/master/docs/testing/python.md#defining-the-ci-test-arguments
+# for details about the block below.
+#
+# === BEGIN CI TEST ARGUMENTS ===
+# test-runner-runs:
+#   run1:
+#     app: ${LIGHTING_APP}
+#     app-args: --discriminator 1234 --KVS kvs1 --trace-to json:${TRACE_APP}.json
+#     script-args: >
+#       --storage-path admin_storage.json
+#       --commissioning-method on-network
+#       --discriminator 1234
+#       --passcode 20202021
+#       --trace-to json:${TRACE_TEST_JSON}.json
+#       --trace-to perfetto:${TRACE_TEST_PERFETTO}.perfetto
+#     factory-reset: true
+#     quiet: true
+# === END CI TEST ARGUMENTS ===
+#
+
+import asyncio
+import logging
+import subprocess
+
+import chip.clusters as Clusters
+import psutil
+from chip.testing.matter_testing import MatterBaseTest, TestStep, async_test_body, default_matter_test_main
+from mobly import asserts
+
+
+def get_pid(name):
+    pid = None
+
+    for proc in psutil.process_iter():
+        if name in proc.name():
+            pid = proc.pid
+            break
+
+    return pid
+
+
+class TC_DGSW_2_2(MatterBaseTest):
+
+    @staticmethod
+    def is_valid_uint64_value(value):
+        return isinstance(value, int) and 0 <= value <= 0xFFFFFFFFFFFFFFFF
+
+    @staticmethod
+    def is_valid_octet_string(value):
+        return isinstance(value, (bytes, bytearray))
+
+    async def send_software_fault_event(self, endpoint, pid):
+       # Construct the FIFO path. The PID should be known beforehand or discovered as part of the setup.
+        fifo_path = f"/tmp/chip_lighting_fifo_{pid}"
```

Comment:
This works only for linux platforms. Can we implement the trigger using a test event trigger as a generic mechanism ?

Reply:
If this avoids the need of using `light` even better because the fewer things we need to compile for CI validation the nicer we are on resource utilization (i.e. faster CI which we want)

Reply:
Yes, we don't need to build light example app with this change